### ISSUE TITLE
Too many conflicts between Firebase External Dependency Manager (1.2.175) and Facebook SDK (15.1.0)

### ISSUE THREAD

**Issue by josealvarez97** [facebook/facebook-sdk-for-unity#667]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-sdk-for-unity/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Unity Editor Version: `2021.3.16f1`
- Facebook SDK for Unity Version: `15.1.0`
- Installation Platform & Version: `iOS` (minimum target version`13`)— Xcode version `14.2`

[Google APIs for Unity](https://developers.google.com/unity/archive) (imported as .tgz tarballs via Unity Package Manager—which is the recommended method by Firebase staff)
- Firebase External Dependency Manager for Unity Version: `1.2.175`
- Firebase App (Core) Version: `10.3.0`
- Google Analytics for Firebase Version: `10.3.0`
- Firebase Cloud Messaging Version: `10.3.0` 

## Goals

What do you want to achieve? 

I want to implement Facebook SDK alongside Firebase External Dependency Manager and other Firebase libraries. 

## Expected Results

What do you expect to happen? 

I would like to implement both libraries without them conflicting with each other.

## Actual Results

What actually happened? Can you provide a stack trace?

My Firebase integration has been working fine for weeks, but the Facebook SDK for Unity conflicts with it because of duplicate DLLs.

(`/Library/PackageCache/` — DLL from Google APIs for Unity `.tgz` tarball)
```
Multiple precompiled assemblies with the same name Google.VersionHandler.dll included on the current platform. Only one assembly with the same name is allowed per platform. (/Users/josealvarez/Documents/GitHub/ReadingApp/Library/PackageCache/com.google.external-dependency-manager@5bc73dfed43e/ExternalDependencyManager/Editor/Google.VersionHandler.dll)
```
(`/Assets/` — DLL from Facebook SDK `.unitypackage` custom package)
```
Multiple precompiled assemblies with the same name Google.VersionHandler.dll included on the current platform. Only one assembly with the same name is allowed per platform. (Assets/PlayServicesResolver/Editor/Google.VersionHandler.dll)
```

In reality, it's not just the Google Version Handler, but all DLLs inside the external dependency manager.

<img width="495" alt="image" src="https://user-images.githubusercontent.com/21250471/214350456-b93136e8-cd75-422f-a3ad-84f10698e4b5.png">

For the record, here are all the compile errors thrown by Unity. They all illustrate the exact same point though.

<img width="1218" alt="image" src="https://user-images.githubusercontent.com/21250471/214351235-3eabc11a-95fe-44fb-9a7b-6a73a0e901c5.png">

<img width="1218" alt="image" src="https://user-images.githubusercontent.com/21250471/214351603-1b2a5eb5-3f15-4377-8c72-f18f686ac298.png">

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/21250471/214351770-46cadba9-adf2-402d-80df-30ad34998284.png">

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/21250471/214351911-f505405c-b965-4c69-9c52-0820f3b1885b.png">

This is how my `/Library/PackageCache` directory would look if I remove the duplicate DLLs:

<img width="841" alt="image" src="https://user-images.githubusercontent.com/21250471/214352551-4a2036f9-050d-4225-8c34-70d85671b066.png">

After attempting to remove the duplicate DLLs located in `/Library/PackageCache` (these would be the versions installed via [Google APIs for Unity](https://developers.google.com/unity/archive) which I imported as .tgz tarballs via Unity Package Manager), I get the following error when attempting to build my project.

```
iOS framework addition failed due to a CocoaPods installation failure. This will will likely result in an non-functional Xcode project.

After the failure, "pod repo update" was executed and succeeded. "pod install" was then attempted again, and still failed. This may be due to a broken CocoaPods installation. See: https://guides.cocoapods.org/using/troubleshooting.html for potential solutions.

pod install output:



    [33mWARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
    Consider adding the following to ~/.profile:

    export LANG=en_US.UTF-8
    [0m
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/unicode_normalize/normalize.rb:141:in `normalize': Unicode Normalization not appropriate for ASCII-8BIT (Encoding::CompatibilityError)
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:166:in `unicode_normalize'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:166:in `installation_root'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:226:in `podfile_path'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/user_interface/error_report.rb:105:in `markdown_podfile'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/user_interface/error_report.rb:30:in `report'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:66:in `report_error'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:396:in `handle_exception'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:337:in `rescue in run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:324:in `run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:52:in `run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/bin/pod:55:in `<top (required)>'
	from /Users/josealvarez/.gem/ruby/2.6.0/bin/pod:23:in `load'
	from /Users/josealvarez/.gem/ruby/2.6.0/bin/pod:23:in `<main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/unicode_normalize/normalize.rb:141:in `normalize': Unicode Normalization not appropriate for ASCII-8BIT (Encoding::CompatibilityError)
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:166:in `unicode_normalize'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:166:in `installation_root'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:226:in `podfile_path'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/config.rb:205:in `podfile'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:160:in `verify_podfile_exists!'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command/install.rb:46:in `run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:52:in `run'
	from /Users/josealvarez/.gem/ruby/2.6.0/gems/cocoapods-1.11.3/bin/pod:55:in `<top (required)>'
	from /Users/josealvarez/.gem/ruby/2.6.0/bin/pod:23:in `load'
	from /Users/josealvarez/.gem/ruby/2.6.0/bin/pod:23:in `<main>'



pod repo update output:

Updating spec repo `trunk`


    [33mWARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
    Consider adding the following to ~/.profile:

    export LANG=en_US.UTF-8
    [0m

UnityEngine.Debug:LogError (object)
Google.Logger:Log (string,Google.LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/Logger.cs:136)
Google.IOSResolver:Log (string,bool,Google.LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/IOSResolver/src/IOSResolver.cs:1159)
Google.IOSResolver:OnPostProcessInstallPods (UnityEditor.BuildTarget,string) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/IOSResolver/src/IOSResolver.cs:2567)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)
``` 

I had never encountered this error before, and most workarounds on the internet ([example 1](https://stackoverflow.com/questions/59405671/ios-cocoapods-requires-your-terminal-to-be-using-utf-8-encoding-after-latest), [example 2](https://forum.unity.com/threads/unity-build-error-error-running-cocoapods.457549/), [example 3](https://github.com/CocoaPods/CocoaPods/issues/6333)) have failed to resolve the issue on my system. 

What is relevant for the maintainers of this repo is that this error seems to come from the Google DLLs imported by the Facebook SDK for Unity. That is, the same DLLs that I had to de-duplicate are now causing problems. The latest versions of these DLLs did not cause these problems when imported directly from Google, but they break the project when imported from the Facebook SDK. 

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/21250471/214344375-960c89fc-c371-4b1a-ad9f-29ccc1d3fd6d.png">

Ideally, I should be able to remove the Google DLLs that come from the Facebook SDK for Unity. If I am not mistaken, this is not trivial for `.unitypackage` imports. When I try to remove them, the Facebook SDK complains that is unable to find them. Please, consider following up on this other issue https://github.com/facebook/facebook-sdk-for-unity/issues/430 asking for a version of the Facebook SDK that can be imported as a .tgz tarball via Unity Package Manager. To the best of my understanding, this would avoid conflicts with duplicates DLLs.

The best solution to this problem would be the one that resolves the issue of of duplicate DLLs. The Facebook SDK for Unity should be able to tell whether the Google APIs for Unity are already located in the project, either in `/Library/PackageCache/` or `/Assets/`. A workaround for the Cocoapods issue would not exactly be the best way to go, but I'll appreciate any advice about how I can move forward with my project.

## Steps to Reproduce

What are the steps necessary to reproduce this issue?

1. Import the Firebase libraries described above from [Google APIs for Unity](https://developers.google.com/unity/archive) (imported as .tgz tarballs via Unity Package Manager, which is the recommended method by Firebase staff). (Note: It's important to import first the External Dependency Manager and Firebase App (Core) libraries before importing Analytics and Messaging.)
2. Import the latest version of the Facebook SDK (`15.1.0`). (Note: I have also tried downgrading to version 14 without success.)
3. Remove the duplicated DLLs from `/Library/PackageCache`.
4. Build and run the project for iOS.

You should see an stack trace with the following error 

```
pod repo update output:

Updating spec repo `trunk`


    [33mWARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
    Consider adding the following to ~/.profile:

    export LANG=en_US.UTF-8
    [0m

UnityEngine.Debug:LogError (object)
Google.Logger:Log (string,Google.LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/Logger.cs:136)
Google.IOSResolver:Log (string,bool,Google.LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/IOSResolver/src/IOSResolver.cs:1159)
Google.IOSResolver:OnPostProcessInstallPods (UnityEditor.BuildTarget,string) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/IOSResolver/src/IOSResolver.cs:2567)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)
```

## Code Samples & Details

Here's a [link](https://github.com/The-Reading-Club/FirebaseFacebookSDKConflict) to a public GitHub repository that you can use to reproduce the issue. Please note that you will have to have Git LFS installed given that I placed the tarballs in the Assets folder.

![image](https://user-images.githubusercontent.com/21250471/214362142-e0d909f7-f89c-4828-85f7-99ba20355415.png)

You won't need a ` GoogleService-Info.plist` file or a Facebook app ID to reproduce the conflict between dependencies on your console.

**Comment by josealvarez97** [comment]
Hi @JordiFB, I was wondering if you could help me with this issue, please. Thank you a lot in advance.

**Comment by josealvarez97** [comment]
[This comment](https://github.com/firebase/quickstart-unity/issues/647#issuecomment-615130142
) on issue https://github.com/firebase/quickstart-unity/issues/647 helped me to be able to build my project again. Just delete the `ExternalDependencyManager` and `PlayServicesResolver` folders. Make sure you delete the whole folders, and not just the DLLs inside, or the Facebook SDK will complain.

However, I cannot run my app yet. Now I am facing issue https://github.com/facebook/facebook-sdk-for-unity/issues/659 again (app crashes at launch on iOS device). I had faced it before, see [my comments in such issue](https://github.com/facebook/facebook-sdk-for-unity/issues/659#issuecomment-1400332025).

**Comment by JordiFB** [comment]
@josealvarez97  Could you check the latest version 16.0.0? We reorganized the External Dependencies Manager folder to make easy for you to not import, just in case you have another plugin like Firebase.

Upgrade guide here: https://developers.facebook.com/docs/unity/upgrading-7.x

**Comment by snappedToGrid** [comment]
I'm using version 16.0.0 along with a Firebase module I downloaded a while ago, and I had the same error.
It seems like I solved it by deleting the 1.2.166 folder that came with the Facebook SDK, and keeping 1.2.175 that I believe came with Firebase.

**Comment by JordiFB** [comment]
@snappedToGrid  FB Unity SDK is delivered with External Dependencies manager v1.2.166. If another plugin like Firebase requires a higher version, you can delete it ( delete the 1.2.166 folder ) or not import the 1.2.166 folder when you are importing the plugin into your project.
